calico_containers/tests/st/utils.py

```diff
@@ -38,6 +24,40 @@ def delete_container(name):
     sh.docker.rm("-f", name, _ok_code=[0, 1])
 
 
+def cleanup_inside(name):
+    """
+    Clean the inside of a container by deleting the containers and images within it.
+    """
+    cleanup_inner_containers(name)
+    cleanup_inner_images(name)
+
+
+def cleanup_inner_containers(name):
+    try:
+        inner_containers = docker("exec", "-t", name, "bash", "-c", "docker ps -qa").stdout.split()
+    except (ErrorReturnCode_127, ErrorReturnCode_255):
+        # 127 caused by '"docker": no command found'
+        # 255 caused by '"bash": executable file not found in $PATH'
+        # These happen in the etcd container.
+        pass
+    else:
+        for container in inner_containers:
+            docker("exec", "-t", name, "bash", "-c", "docker rm -f %s" % container)
+
+
+def cleanup_inner_images(name):
+    try:
+        inner_images = docker("exec", "-t", name, "bash", "-c", "docker images -q").stdout.split()
+    except (ErrorReturnCode_127, ErrorReturnCode_255):
+        # 127 caused by '"docker": no command found'
```

Comment:
One last thought: can both of these really occur?  Seems odd that we'd get different errors on different runs.